Sources/StreamChatUI/ChatMessageList/ChatMessageListIndicatorOverlayView.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// An `UIView` subclass indicating that user or multiple users are currently typing.
+public typealias ChatMessageTypingIndicatorOverlayView = _ChatMessageListIndicatorOverlayView<NoExtraData>
+
+/// An `UIView` subclass indicating that user or multiple users are currently typing.
+open class _ChatMessageListIndicatorOverlayView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+    var content: String? {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    /// The animated view with three dots indicating that someone is typing.
+    open private(set) lazy var typingIndicatorView: _ChatTypingIndicatorView<ExtraData> = uiConfig
+        .messageList
+        .typingIndicatorView
+        .init()
+        .withoutAutoresizingMaskConstraints
+
+    /// Label describing who is currently typing
+    /// `User is typing`
+    /// `User and 3 more are typing`
+    open private(set) lazy var informationLabel: UILabel = UILabel().withoutAutoresizingMaskConstraints
+
+    /// StackView holding `typingIndicatorView` and `informationLabel`
+    open private(set) lazy var componentStackView: UIStackView = UIStackView().withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        typingIndicatorView.startAnimating()
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+        embed(componentStackView)
+        componentStackView.addArrangedSubview(typingIndicatorView)
+
+        componentStackView.addArrangedSubview(informationLabel)
+        componentStackView.distribution = .fill
+        componentStackView.alignment = .center
+
+        typingIndicatorView.heightAnchor.pin(equalToConstant: 5).isActive = true
+        typingIndicatorView.centerYAnchor.pin(equalTo: centerYAnchor).isActive = true
+    }
+
+    override public func defaultAppearance() {
+        layer.opacity = 0.8
```

Comment:
This opacity affects also the label and the animation view. Is that intentional?

Reply:
Nope, I actually fixed this with creating new color ( discovered that we are missing color `overlayLight` which is contrast to `overlayDark`), this will be fixed in upcoming commits.